### PR TITLE
Merge from micropython v1.22.2

### PR DESCRIPTION
**Pull request by dhalbert**
Merging from `micropython-v1.22-release` branch.

- added background tasks and interrupt checks to new routines `mp_event_wait_ms() `and `mp_event_wait_indefinite()`.
- Note that `QDEF` is now `QDEF0` and `QDEF1`.
- `pyexec.c` is still hard to merge due to `atexit` conditionals. Maybe there's some way of formatting them so there are fewer changes.
- `raspberrypi/libm_extra.c `fixes ROM floats to handle infinities correctly, but I did not pick this up yet.

As of the initial draft PR, at least one board from each port has been compiled. `atmel-samd`, `nordic,`espressif`, `rasbperrypi` have had some basic testing done, blink for most, BLE for nordic, and wifi for `espressif`. I'm submitting the PR to see how the tests and doc builds work out.

## Update: out of draft mode.

Everything builds except for two too-large builds, which I will fix soon, either here or in a later PR. Smoke tests work. 

- I added quite a few more `CIRCUITPY-CHANGE` annotations. They are more or less complete now.
- I updated the https://github.com/adafruit/Adafruit_CircuitPython_asyncio library. The changes are currently in a branch: https://github.com/adafruit/Adafruit_CircuitPython_asyncio/tree/v1.22-merge. That branch is what is selected for `frozen/Adafruit_CircuitPython_asyncio` at the moment. I need to test it against 9.1.1 and see if the changes still work for 9.1.1 before I merge it into main in the library. So anyone using `asyncio` with this build should use that branch for now.

### CODE REVIEW

**Comment by dhalbert** [comment]
Ready for review. There are two builds failing due to space issues. This is a bit surprising, but I don't think this should hold up reviewing the whole thing. I'm thinking there may be some extra math code or something like that. I will look and see what grew.

**Comment by dhalbert** [comment]
[also put in OP]

## Update: out of draft mode.

Everything builds except for two too-large builds, which I will fix soon, either here or in a later PR. Smoke tests work. 

- I added quite a few more `CIRCUITPY-CHANGE` annotations. They are more or less complete now.
- I updated the https://github.com/adafruit/Adafruit_CircuitPython_asyncio library. The changes are currently in a branch: https://github.com/adafruit/Adafruit_CircuitPython_asyncio/tree/v1.22-merge. That branch is what is selected for `frozen/Adafruit_CircuitPython_asyncio` at the moment. I need to test it against 9.1.1 and see if the changes still work for 9.1.1 before I merge it into main in the library. So anyone using `asyncio` with this build should use that branch for now.

**Comment by dhalbert** [comment]
@tannewt all green now

File: py/objint.c
Comment (tannewt) [inline py/objint.c:533]:
```suggestion
// CIRCUITPY-CHANGE: supports signed
```

File: py/qstr.h
Comment (tannewt) [inline py/qstr.h:116]: Want to do this?

File: shared-bindings/rainbowio/__init__.h
Comment (tannewt) [inline shared-bindings/rainbowio/__init__.h:12]:
```suggestion
```

File: tools/merge_micropython.py
Comment (tannewt) [inline tools/merge_micropython.py:50]: Want to commit this?

**Review by tannewt — CHANGES_REQUESTED** [review]
Thanks for doing this! Just a few minor things. Mostly typos

File: py/mkrules.mk
Comment (dhalbert) [reply py/mkrules.mk:171]: It is weird this didn't cause an error!

File: py/qstr.h
Comment (dhalbert) [reply py/qstr.h:116]: I did :), and forgot to remove the comment.

File: tools/merge_micropython.py
Comment (dhalbert) [reply tools/merge_micropython.py:50]: Oops. I run pieces of this at a time.

**Review by tannewt — APPROVED** [review]
Looks good! Thank you!

**Pull request merged**
